web-local/src/lib/svelte-query/actions.ts

```diff
@@ -0,0 +1,83 @@
+import { goto } from "$app/navigation";
+import {
+  getRuntimeServiceListFilesQueryKey,
+  V1DeleteFileAndReconcileResponse,
+  V1RenameFileAndReconcileResponse,
+} from "@rilldata/web-common/runtime-client";
+import type { ActiveEntity } from "@rilldata/web-local/common/data-modeler-state-service/entity-state-service/ApplicationEntityService";
+import type { EntityType } from "@rilldata/web-local/common/data-modeler-state-service/entity-state-service/EntityStateService";
+import { getNextEntityName } from "@rilldata/web-local/common/utils/getNextEntityId";
+import { dataModelerService } from "@rilldata/web-local/lib/application-state-stores/application-store";
+import { commonEntitiesStore } from "@rilldata/web-local/lib/application-state-stores/common-store";
+import {
+  getFileFromName,
+  getLabel,
+  getRouteFromName,
+} from "@rilldata/web-local/lib/components/entity-mappers/mappers";
+import notifications from "@rilldata/web-local/lib/components/notifications";
+import { queryClient } from "@rilldata/web-local/lib/svelte-query/globalQueryClient";
+import type { UseMutationResult } from "@sveltestack/svelte-query";
+
+export async function renameEntity(
+  instanceId: string,
+  fromName: string,
+  toName: string,
+  type: EntityType,
+  renameMutation: UseMutationResult<V1RenameFileAndReconcileResponse>
+) {
+  const resp = await renameMutation.mutateAsync({
```

Comment:
Whenever I see `await mutation.mutateAsync()`, I keep thinking it's some purposeful hack to achieve functionality that `mutation.mutate()` doesn't offer. I'd much prefer we go with the out-of-the-box synchronous approach `mutation.mutate()` to keep it simple.

Reply:
I am not sure I agree here. Is using mutate the standard? It goes leads to callbacks which is what async/await was added to remove.

Reply:
I like this blog for best practices, and he actually discusses exactly this `mutate` or `mutateAsync`, and recommends `mutate`: https://tkdodo.eu/blog/mastering-mutations-in-react-query#mutate-or-mutateasync

The docs also have a section on when `mutateAsync()` would be useful, but the motivation seems to be different from what we have currently implemented: https://tanstack.com/query/v4/docs/guides/mutations#promises

Reply:
While I do agree that article has some really good points. It is not a silver bullet. In our particular use case where we have common methods outside of component async makes more sense. Otherwise we would need to add callbacks to rename and delete here. Which would take us back to callback issue in javascript pre async-await.